lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -57,15 +58,17 @@ defmodule Mix.Compilers.Elixir do
         new_paths =
           all_paths
           |> MapSet.difference(prev_paths)
-          |> MapSet.to_list
+          |> MapSet.to_list()
 
         # Plus the sources that have changed in disk
-        for(source(source: source, external: external, size: size) <- all_sources,
-            {last_mtime, last_size} = Map.fetch!(sources_stats, source),
-            times = Enum.map(external, &(sources_stats |> Map.fetch!(&1) |> elem(0))),
-            size != last_size or Mix.Utils.stale?([last_mtime | times], [modified]),
-            into: new_paths,
-            do: source)
+        for(
+          source(source: source, external: external, size: size) <- all_sources,
```

Comment:
I tried looking for a function definition for `source()` but didn't manage to find anything. how does this part work?

it seems similar to a destructuring assignment, like `const { source, external, size } = object` in ES6, but if it was that, would you be able to do `{source: source, external: external, size: size} <- all_sources` instead?

Reply:
ahhh, looks like it's a macro?
```
iex(9)> h Mix.Compilers.Elixir.source
source/0    source/1    source/2
iex(9)> h Mix.Compilers.Elixir.source/0
                          defmacro source(args \\ [])
iex(10)> h Mix.Compilers.Elixir.source/1
                          defmacro source(args \\ [])
```